internal/controllers/operator_controller.go

```diff
@@ -472,7 +466,12 @@ func operatorRequestsForCatalog(c client.Reader, logger logr.Logger) handler.Map
 // and joins them with a semicolon (rather than a comma, which the unsat.Error()
 // function does). This function also has the side effect of sorting the items
 // in the unsat slice.
-func prettyUnsatMessage(unsat deppy.NotSatisfiable) string {
+func prettyUnsatMessage(err error) string {
+	unsat := deppy.NotSatisfiable{}
+	if !errors.As(err, &unsat) {
+		panic("we must always receive deppy.NotSatisfiable here")
```

Comment:
I'm always hesitant about introducing panics in code, could you explain the benefit?

Reply:
`prettyUnsatMessage` is a workaround https://github.com/operator-framework/deppy/issues/142 (there is a comment about it above, not in the diff). It will go away and we will be doing something like this in controller code eventually:

```go
setResolvedStatusConditionFailed(&op.Status.Conditions, err.Error(), op.GetGeneration())
```

I was thinking failing laudly given that this is a temporary code and this condition should never happen unless we introduce a regression. But now I'm thinking about it again - yes, probably there is not much benefit in this and we can do just something like this: 

```go
unsat := deppy.NotSatisfiable{}
if !errors.As(err, &unsat) {
	return err.Error()
}
```

I'll update.

Reply:
Done